### ISSUE TITLE
Revert making drop_attached_node local

### ISSUE THREAD

**Issue by CasimirKaPazi** [minetest/minetest#5665]
For my subgame I override drop_attached_node to make nodes falling instead of dropping. With 649448a this behavior broke, because the function was made local.

I could either override the function(s) that call drop_attached_node, or we could make it public like core.check_for_falling. I think the later is cleaner, and there is no reason to keep the function local.

______________

Also btw. it seems that check for falling is unsymmetrical. The first four lines are out of place.

    local check_for_falling_neighbors = {
    	{x = -1, y = -1, z = 0},
    	{x = 1, y = -1, z = 0},
    	{x = 0, y = -1, z = -1},
    	{x = 0, y = -1, z = 1},
    	{x = 0, y = -1, z = 0},
    	{x = -1, y = 0, z = 0},
    	{x = 1, y = 0, z = 0},
    	{x = 0, y = 0, z = 1},
    	{x = 0, y = 0, z = -1},
    	{x = 0, y = 0, z = 0},
    	{x = 0, y = 1, z = 0},
    }

**Comment by sfan5** [comment]
I think `drop_attached_node` and others being non-local was unintended.

**Comment by paramat** [comment]
The order of `check_for_falling_neighbors` was very carefully designed by sofar.

**Comment by sofar** [comment]
Actually I made them `local` on purpose, they were undocumented APIs and I didn't see or find any actual users outside of `builtin` at the time.

**Comment by sofar** [comment]
The order of the `check_for_falling_neighbors` table is intentional, we want to make sure that we check nodes *downward* first otherwise you get non-uniformly falling and possibly mid-air jammed nodes.

Note that we wouldn't have to have any specific ordering if the table contained all 26 neighbors, but that would cause a lot of duplicate checks, and the table method above optimizes specifically for the wanted effect (which is, everything falls at the same time as much as possible) while at the same time checking as few neighbors as possible to get the wanted effect.

The diagonal down checks exists to find the lowest node in the falling chain as fast as possible, basically.

**Comment by CasimirKaPazi** [comment]
> and I didn't see or find any actual users outside of builtin at the time.

Well I am one. However I *am* able to override the function that call drop_attached_node. So it's not that I am not able to use it, I just found it to be ugly that I had to copy three functions with about 65 additional lines of code, completely unaltered, just to change one line of code in another function. For something that worked fine all the time before.
